### DISCUSSION TITLE
Any way to force pnpm workspace to always use the same version + install of a package?

### DISCUSSION THREAD

**Discussion by ian** [pnpm/pnpm#5532] [Q&A]
We're running into lots of issues around pnpm desire to install multiple versions of the same package based on how it resolves (perceived) dependencies.

This is particularly an issue with `wagmi`, since it uses an internal variable to store the config/client, and having multiple versions of wagmi will fail to load this config. I'd imagine this is quite common.

Consider the scenario where you have apps/api and apps/nextjs, both which import the wagmi package (or ethers for that matter). Even though they both resolve to 0.7.5, you will get multiple 0.7.5 installs with a hash signature. 

Is it possible (through a .npmrc or equivalent) to force pnpm workspace to ALWAYS USE THE SAME PACKAGE INSTALL?

Have tried to use pnpm overrides but this simply forces a specific package version, it doesn't force pnpm to use the SAME package install.

**Comment by ghost** [comment]
@ian did you get any answer on this? I am sure there is some hosting-option or something, but I cannot find the right one.

**Reply by zkochan** [reply]
There is no such option but since the question was asked the chance of duplication has been reduced with the [dedupe-peer-dependents](https://pnpm.io/npmrc#dedupe-peer-dependents) setting that is "true" by default.

**Reply by janis-me** [reply]
Thanks for the reply! Good to know. I ended up having to use `packageExtensions` to override optionalDependencies. But it was a weird use-case for me anyways.
